### PR TITLE
initial commit for the action

### PR DESCRIPTION
**Pull request by walteck**
## Description
The initial commit - copying the format from [actions/typescript-action](https://github.com/actions/typescript-action) to implement the framework plus updating main.ts to enable the sending of a notification to MS Teams.

## Context

This is to enable teams to easily send notifications from their build pipelines to Microsoft Teams.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [X] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [X] I have followed the code style of the project
- [^] I have added tests to cover my changes
- [X] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [X] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          npm install
+      - run: |
+          npm run all
+
+      - uses: ./
+        with:
+          github-token: ${{ github.token }}
```
Comment (jacobgill1) [inline .github/workflows/test.yml:21]: We usually access the github token like this: `${{ secrets.GITHUB_TOKEN }}`
I'm not sure if it'd be different in this context though

Comment (walteck) [reply .github/workflows/test.yml]: thanks I've updated.

File: Makefile [hunk 2]
```diff
@@ -3,14 +3,21 @@ include ./scripts/init.mk
 # This file contains hooks into the project configuration, test and build cycle
 # as automated steps to be executed on a workstation and in the CI/CD pipeline.
 
+install: # Install project dependencies
+	npm install
+
+build: # Build project
+	npm run build && npm run package
```
Comment (stefaniuk) [inline Makefile:10]: Could we break this line down to two to improve readability? 

```
npm run build
npm run package
```

File: Makefile [hunk 3]
```diff
@@ -3,14 +3,21 @@ include ./scripts/init.mk
...
+
 config: # Configure development environment
-	# TODO: Use only `make` targets that are specific to this project, e.g. you may not need to install Node.js
 	make \
-		asdf-install \
-		githooks-install \
-		nodejs-install \
-		python-install \
-		terraform-install
+		nodejs-install
```
Comment (stefaniuk) [inline Makefile:17]: Could we add a call to the other two make targets `asdf-install` and  `githooks-install` to the config recipe? E.g.

```
make \
    asdf-install \
    githooks-install \
    nodejs-install
```

File: .github/workflows/check-dist.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set Node.js 16.x
+        uses: actions/setup-node@v3.7.0
+        with:
+          node-version: 16.x
+
+      - name: Install dependencies
+        run: npm ci
```
Comment (stefaniuk) [inline .github/workflows/check-dist.yml:32]: Could we call `make install` instead of `npm ci`?

File: .github/workflows/check-dist.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: Rebuild the dist/ directory
+        run: |
+          npm run build
```
Comment (stefaniuk) [inline .github/workflows/check-dist.yml:36]: For consistency, could we collapse these two commands and run `make build` instead?

**Review by stefaniuk — COMMENTED** [review]
Some minor suggestions for consideration.

File: Makefile (hunk 2 above)
Comment (walteck) [reply Makefile:10]: Updated

File: .github/workflows/check-dist.yml (hunk 4 above)
Comment (walteck) [reply .github/workflows/check-dist.yml:32]: done

**Review by stefaniuk — APPROVED** [review]
LGTM

**Pull request merged**